src/modules/paymentProcessor/PP_Connext_Crosschain_v1.sol

```diff
@@ -1,117 +1,349 @@
+            revert Module__PP_Crosschain__InvalidUnclaimableAmount();
+        }
+
+        // Subtract the order amount from the unclaimable amount.
+        _unclaimableAmountsForRecipient[client_][order_.paymentToken][recipient_]
+        -= order_.amount;
+
+        // Delete the unclaimable amount mapping entry if the amount is 0.
+        if (
+            _unclaimableAmountsForRecipient[client_][order_.paymentToken][recipient_]
+                == 0
+        ) {
+            delete _unclaimableAmountsForRecipient[client_][order_.paymentToken][recipient_];
+        }
 
-            // Inform the client about the processed amount
-            client.amountPaid(orders[i].paymentToken, orders[i].amount);
+        // Execute the bridge transfer.
+        bytes memory bridgeData = _executeBridgeTransfer(order_);
+
+        // If a zero intent ID is returned, revert.
+        if (bytes32(bridgeData) == bytes32(0)) {
+            revert Module__PP_Crosschain__MessageDeliveryFailed(
+                order_.originChainId,
+                order_.targetChainId,
+                order_.flags,
+                order_.data
+            );
         }
+        // Store the intent ID for the payment order.
+        _bridgeData[_paymentId] = bridgeData;
+        _paymentId++;
+    }
+
+    /// @inheritdoc IPaymentProcessor_v1
+    function validPaymentOrder(
+        IERC20PaymentClientBase_v2.PaymentOrder memory order
+    ) external virtual returns (bool valid_) {
+        return _validPaymentOrder(order);
+    }
+
+    // -------------------------------------------------------------------------
+    // Internal Functions
+
+    /// @notice Validates the payment order.
+    /// @param order The payment order to validate.
+    /// @return valid_ True if the payment order is valid, false otherwise.
+    function _validPaymentOrder(
+        IERC20PaymentClientBase_v2.PaymentOrder memory order
+    ) internal virtual returns (bool valid_) {
+        valid_ = _validPaymentReceiver(order.recipient)
+            && _validPaymentToken(order.paymentToken) && _validTotal(order.amount)
+            && _validateOriginAndTargetChainId(
+                order.originChainId, order.targetChainId
+            ) && _validateFlagsAndData(order.flags, order.data);
+
+        (uint48 maxFee, uint48 ttl) =
+            _getEverclearMaxFeeAndTTL(order.flags, order.data);
+        bool validParams = maxFee > 0 && ttl > 0;
+
+        return validParams && valid_;
     }
 
-    function xcall(
-        IERC20PaymentClientBase_v1.PaymentOrder memory order,
-        bytes memory executionData
-    ) internal returns (bytes32 intentId) {
-        // Decode any additional parameters from executionData
-        (uint maxFee, uint ttl) = abi.decode(executionData, (uint, uint));
+    /// @dev Execute the cross-chain bridge transfer
+    /// @param order The payment order containing transfer details
+    /// @return bridgeData Data returned by the bridge implementation
```

Comment:
```suggestion
    /// @notice Execute the cross-chain bridge transfer
    /// @param  order_ The payment order containing transfer details
    /// @return bridgeData_ Data returned by the bridge implementation
```

Reply:
Inverter Standard:
- All function should have @notice tag. @dev tag is only added if additional information is added specfic for development
- Tab between the natspec tag and comment